### PR TITLE
Clarify constraints and examples in ug

### LINKED ISSUE #325: [PE-D][Tester B] Rate parameter error message not specific
The error message for the invalid use of the `rate` parameter as `r/hello` does not match the error. It is also not mentioned in the UG that `rate` must be a number.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: SQ77/ped#8

### LINKED ISSUE #326: [PE-D][Tester E] Extraneous parameters for all commands with no parameters not working as specified
Similar to another bug i had reported, the extraneous parameters may not be parsed as intended for the commands listed in the user guide.

Perhaps more examples could be provided to show what is accepted and what is not.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: seantham21/ped#2

### LINKED ISSUE #337: [PE-D][Tester D] More rigorous input handling for schedule
When inputting add command `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01 t/Tuesday-1300- 1500 s/economics r/50.00 paid/100.00 owed/0.00`, the whitespace in schedule format throws an error.

While constraint is specified in the error message, it is not outright specified in the UG. I suggest for the team to either state it in the UG or (recommend) to allow for more rigorous input handling to allow for whitespace as fast typists may accidentally input a whitespace.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: fymbc/ped#4

### LINKED ISSUE #350: [PE-D][Tester E] Edit paid/owed does not account for -0
Should the value -0 be specified as a constraint? As the value is still 0 nonetheless.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: seantham21/ped#11

### LINKED ISSUE #353: [PE-D][Tester E] Command Parser does not account for missing whitespaces
As seen here, should such an input be allowed, or should the number of whitespaces be strictly adhered to? I think this should be explicitly mentioned in the User Guide.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: seantham21/ped#12

### LINKED ISSUE #359: [PE-D][Tester C] UG: Unclear notation used in the find command
It is unclear (to me at least) that [MORE_DAYS...] means that I can add a space separated list of keywords, only realized this through testing the inputs for this find command.

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: YangQF2002/ped#9

### PR DESCRIPTION
**Pull request by Feng1231**
Changes:
* Add more example for `find` command, closes #359 
* Inform user for special case of negative `0`, closes #350 
* Inform user that schedule should not 
contain any space in between, closes #337 
* Inform user that extra parameters for
commands that take in no parameters should 
come after the command word, closes #326 
* Inform user that rate must be a number, closes #325
* Inform user that the command word and each of its parameters should be separated by at least one space, closes #353

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -140,15 +143,18 @@ Adds a student to the address book.
...
 * **START_TIME** and **END_TIME** are represented as `HHmm`.
 * **PHONE_NUMBER** should be 8 digits that starts with 6, 8 or 9.
 * **RATE** is the tuition fee per hour. It must meet the following criteria:
+  * It is a positive numeric value with at most 2 decimal places.
   * Minimum: $0.01 (must be a positive value)
   * Maximum: $1000.00 (two decimal places allowed)
+  > Reason for the maximum value: It is unlikely for an undergraduate tutor to have an hourly rate higher than $1000.00.
 * **PAID_AMOUNT** and **OWED_AMOUNT** must be at least 0 with at most 2 decimal places.
       <i>Example: </i> `12.00`, `0.0` or `7`.
+  * Special case: `-0`, `-0.0`, `-0.00` are not allowed!
```
Comment (Soumyadip-cmd) [inline docs/UserGuide.md:157]: Maybe can add in that this is because it will cause the system to act wierdly

**Review by Soumyadip-cmd — COMMENTED** [review]
Some considerations

File: docs/UserGuide.md (hunk 1 above)
Comment (Feng1231) [reply docs/UserGuide.md:157]: Should I create a known issue section for this?

Comment (DennieDan) [reply docs/UserGuide.md:157]: Good idea

**Review by DennieDan — COMMENTED** [review]
We are not allowed to change the MESSAGE_USAGE of Rate, right?

image


I think it is allowed as we don't change how rate is validated, we just add in more word ("must be a number")

**Comment by Feng1231** [comment]
> We are not allowed to change the MESSAGE_USAGE of Rate, right?
> 
> 
> 
> image
> 
> 
> 
> 
> 
> I think it is allowed as we don't change how rate is validated, we just add in more word ("must be a number") 

Cannot make any changes to message usage.
Unless it makes it more general

Anyway i think that tester is just nit picking

**Review by DennieDan — APPROVED** [review]

**Review by Soumyadip-cmd — APPROVED** [review]
LGTM

**Pull request merged**